python/cudf/cudf/core/column/column.py

```diff
@@ -77,26 +78,16 @@ def data_array_view(self):
 
         if is_categorical_dtype(self.dtype):
             return self.codes.data_array_view
-        else:
-            dtype = self.dtype
 
-        result = rmm.device_array_from_ptr(
-            ptr=self.data.ptr, nelem=len(self), dtype=dtype
-        )
-        result.gpu_data._obj = self
+        result = cp.asarray(self.data)
```

Comment:
It seems like `cp.asarray` is called more often in the codebase than `cuda.as_cuda_array`. So this may not need a justification. That said, I did a quick benchmark (results below) and found it faster to coerce to a CuPy array. Though we were using Numba here. So not sure how important this is. Push back is welcome.

```python
In [1]: import numpy                                                            

In [2]: import cupy                                                             

In [3]: import numba.cuda                                                       

In [4]: import rmm                                                              

In [5]: rmm.reinitialize(pool_allocator=True, 
   ...:                  initial_pool_size=int(2 * 2**30))                      
Out[5]: 0

In [6]: rdb = rmm.DeviceBuffer(size=50_000_000)                                 

In [7]: %timeit numba.cuda.as_cuda_array(rdb)                                   
46.3 µs ± 603 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [8]: %timeit cupy.asarray(rdb)                                               
3.5 µs ± 42.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Also should add that switching from `rmm.device_array_from_ptr` to `cupy.asarray` significantly improves performance. Taking a `rmm.DeviceBuffer` as an example, this is roughly an order of magnitude faster.

```python
In [1]: import cupy                                                             

In [2]: import rmm                                                              

In [3]: rmm.reinitialize(pool_allocator=True, 
   ...:                  initial_pool_size=int(2 * 2**30))                      
Out[3]: 0

In [4]: rdb = rmm.DeviceBuffer(size=50_000_000)                                 

In [5]: %timeit rmm.device_array_from_ptr(rdb.ptr, nelem=rdb.size, dtype="u1")  
27 µs ± 81 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)

In [6]: %timeit cupy.asarray(rdb)                                               
2.94 µs ± 13.5 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Note: `rmm.device_array_from_ptr` is quite a bit faster than `numba.cuda.as_cuda_array` as it interfaces with some lower-level pieces in Numba's machinery thus avoiding some overhead that would otherwise be seen.

Reply:
@jakirkham sounds good to me. We just need to do a quick pass and fix docstrings where it says they return numba device arrays instead of cupy arrays.

Reply:
Have updated the docstrings in this file at least. Please let me know if there are others we should look at 🙂